internal/cli/status.go

```diff
@@ -146,6 +156,458 @@ func (c *StatusCommand) Run(args []string) int {
+}
+
+// refreshAppStatusLocal takes a project and a single application and uses
+// the local runner to execute a StatusReport refresh function by getting
+// the latest Deployment or Release and using the application client to
+// refresh the apps Status Report.
+func (c *StatusCommand) refreshAppStatusLocal(
+	project *pb.Project,
+	app *pb.Application,
+	workspace string,
+) error {
+	// Get our API client
+	client := c.project.Client()
+
+	// Get Latest Deployment and Release
+
+	// Deployments
+	deploymentsResp, err := client.UI_ListDeployments(c.Ctx, &pb.UI_ListDeploymentsRequest{
+		Application: &pb.Ref_Application{
+			Application: app.Name,
+			Project:     project.Name,
+		},
+		Workspace: &pb.Ref_Workspace{
+			Workspace: workspace,
+		},
+		Order: &pb.OperationOrder{
+			Order: pb.OperationOrder_COMPLETE_TIME,
+			Limit: 1,
+		},
+	})
+	if err != nil {
+		if s, ok := status.FromError(err); ok {
+			if s.Code() == codes.Unimplemented {
+				return c.errAPIUnimplemented(err)
+			}
+		}
+		return err
+	}
+
+	var deployment *pb.Deployment
+	if deploymentsResp.Deployments != nil && len(deploymentsResp.Deployments) > 0 {
+		deployment = deploymentsResp.Deployments[0].Deployment
+	} else {
+		c.ui.Output("No deployments in project %q for app %q to refresh a status on!",
+			project.Name,
+			app.Name,
+			terminal.WithWarningStyle())
+		// We probably don't have a release either, so return early
+		return nil
+	}
+
+	// Get our app client
+	appClient := c.project.App(app.Name)
+
+	_, err = appClient.StatusReport(c.Ctx, &pb.Job_StatusReportOp{
+		Target: &pb.Job_StatusReportOp_Deployment{
+			Deployment: deployment,
+		},
+	})
+	if err != nil {
+		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return err
+	}
+
+	// Releases
+	releaseResp, err := client.UI_ListReleases(c.Ctx, &pb.UI_ListReleasesRequest{
+		Application: &pb.Ref_Application{
+			Application: app.Name,
+			Project:     project.Name,
+		},
+		Workspace: &pb.Ref_Workspace{
+			Workspace: workspace,
+		},
+		Order: &pb.OperationOrder{
+			Order: pb.OperationOrder_COMPLETE_TIME,
+			Limit: 1,
+		},
+	})
+	if err != nil {
+		if s, ok := status.FromError(err); ok {
+			if s.Code() == codes.Unimplemented {
+				return c.errAPIUnimplemented(err)
+			}
+		}
+		return err
+	}
+
+	var release *pb.Release
+	if releaseResp.Releases != nil && len(releaseResp.Releases) > 0 {
+		release = releaseResp.Releases[0].Release
+	}
+
+	if release != nil && !release.Unimplemented {
+		_, err = appClient.StatusReport(c.Ctx, &pb.Job_StatusReportOp{
+			Target: &pb.Job_StatusReportOp_Release{
+				Release: release,
+			},
+		})
+		if err != nil {
+			c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+			return err
+		}
+	}
+
+	return nil
+}
+
+// refreshAppStatusRemote takes a single project and application and uses a
+// remote runner through the Waypoint Server API to queue and start a
+// status report refresh job. It takes the queued job id and hooks into the
```

Comment:
Does this need to be performed by a remote runner? I think it would be valuable for projects without a remote runner configured to be able to do this too.

Reply:
See this convo! https://github.com/hashicorp/waypoint/pull/2081#discussion_r694087217 I think this is essentially your question